lib/Providers/OpenIDConnectProvider.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace SimpleSAML\Module\authoauth2\Providers;
+
+use Firebase\JWT;
+use League\OAuth2\Client\Provider\AbstractProvider;
+use League\OAuth2\Client\Provider\Exception\IdentityProviderException;
+use League\OAuth2\Client\Provider\GenericResourceOwner;
+use League\OAuth2\Client\Tool\BearerAuthorizationTrait;
+use League\OAuth2\Client\Token\AccessToken;
+use Psr\Http\Message\ResponseInterface;
+use SimpleSAML\Logger;
+use SimpleSAML\Utils\HTTP;
+
+class OpenIDConnectProvider extends AbstractProvider
+{
+    use BearerAuthorizationTrait;
+
+    const CONFIGURATION_PATH = '/.well-known/openid-configuration';
+    const ERROR_FIELD = 'error';
+
+    /**
+     * @var string
+     */
+    protected $issuer;
+
+    /**
+     * @var array
+     */
+    private $openIdConfiguration;
+
+    /**
+     * @var string
+     */
+    private $responseError = 'error';
+
+    public function __construct(array $options = [], array $collaborators = [])
+    {
+        parent::__construct($options, $collaborators);
+
+        if (!array_key_exists('issuer', $options))
+        {
+            throw new \InvalidArgumentException(
+                'Required options not defined: issuer'
+            );
+        }
+        $this->issuer = $options['issuer'];
+    }
+
+    protected function getScopeSeparator()
+    {
+        return ' ';
```

Comment:
Could this be done with a class constant?

Reply:
This implements an interface from the upstream oauth2-client library. We could of course define a class constant and return that here, but I don't see the point of that. To replace the entire function with a constant would require api changes to the library we are using

Reply:
If it's an external library, nevermind. Just not worth it.